packages/wasm/package.json

```diff
@@ -30,12 +30,154 @@
-    "bindgen": "wasm-bindgen --weak-refs --target $TARGET --out-dir dist/$TARGET target/wasm32-unknown-unknown/release/wasm.wasm",
-    "opt": "wasm-opt -O4 dist/$TARGET/wasm_bg.wasm -o dist/$TARGET/wasm_bg.wasm",
-    "build": "cross-env TARGET=nodejs pnpm run target && move-file dist/nodejs/wasm.js dist/nodejs/wasm.cjs && cross-env TARGET=web pnpm run target && cpy --flat src/workerd.js dist/web && cross-env TARGET=bundler pnpm run target",
-    "test": "pw-test --mode node && pw-test"
+    "build": "export PROFILE=dev && export TARGET_DIR=debug && pnpm run build:all",
+    "release": "export PROFILE=release && export TARGET_DIR=release && pnpm run build:all",
+    "build:all": "wireit",
+    "clean": "wireit",
+    "test": "wireit",
+    "test:node": "wireit"
+  },
+  "wireit": {
+    "compile": {
+      "command": "cargo build --target wasm32-unknown-unknown --profile $PROFILE",
+      "env": {
+        "PROFILE": {
+          "external": true
+        }
+      }
+    },
+    "opt": {
+      "command": "wasm-opt -O4 target/wasm32-unknown-unknown/$TARGET_DIR/wasm.wasm -o target/wasm32-unknown-unknown/$TARGET_DIR/wasm.wasm",
+      "env": {
+        "TARGET_DIR": {
+          "external": true
+        }
+      },
+      "dependencies": [
+        "compile"
+      ]
+    },
+    "bindgen:bundler": {
+      "command": "wasm-bindgen --weak-refs --target bundler --out-dir dist/bundler target/wasm32-unknown-unknown/$TARGET_DIR/wasm.wasm",
+      "env": {
+        "TARGET_DIR": {
+          "external": true
+        }
+      },
+      "dependencies": [
+        "opt"
+      ],
+      "output": [
+        "dist/bundler"
+      ]
+    },
+    "bindgen:nodejs": {
+      "command": "wasm-bindgen --weak-refs --target nodejs --out-dir dist/nodejs target/wasm32-unknown-unknown/$TARGET_DIR/wasm.wasm && move-file dist/nodejs/wasm.js dist/nodejs/wasm.cjs",
+      "env": {
+        "TARGET_DIR": {
+          "external": true
+        }
+      },
+      "dependencies": [
+        "opt"
+      ],
+      "output": [
+        "dist/nodejs"
+      ]
+    },
+    "bindgen:web": {
+      "command": "wasm-bindgen --weak-refs --target web --out-dir dist/web target/wasm32-unknown-unknown/$TARGET_DIR/wasm.wasm && cpy --flat src/workerd.js dist/web",
+      "env": {
+        "TARGET_DIR": {
+          "external": true
+        }
+      },
+      "dependencies": [
+        "opt"
+      ],
+      "output": [
+        "dist/web"
+      ]
+    },
+    "build:all": {
+      "dependencies": [
+        "bindgen:bundler",
+        "bindgen:nodejs",
+        "bindgen:web"
+      ]
+    },
+    "clean": {
+      "command": "rimraf dist"
+    },
+    "test:prepare": {
+      "command": "npx playwright install && cross-env mkdir test/report",
+      "output": [
+        "test/report"
+      ]
+    },
+    "test:chromium": {
+      "command": "pw-test test/wasm.test.js -r mocha --reporter json --cov > test/report/chromium.json",
```

Comment:
Was considering adding the `--cov` for code coverage to `test:node` so a developer could get coverage reports during development without testing all environments. `playwright-test` reports an error when I move it there, so left it here for now.

Reply:
pw-test fails in node mode with --cov ?

Reply:
Yep, I think so. This command emits no output, where a JSON report is expected:

```sh
npx playwright-test test/wasm.test.js -r mocha --reporter json --mode node --cov
```

Also, `wireit` reports an exit status 1 for `test:node`.